components/AppBar/src/MDCAppBarNavigationController.m

```diff
@@ -36,7 +36,7 @@ @implementation MDCAppBarNavigationControllerInfo
 - (void)dealloc {
   // On pre-iOS 11 devices we need to manually clear out the trackingScrollView because we've
   // enabled the observesTrackingScrollViewScrollEvents behavior on the flexible header.
-  self.appBar.headerViewController.headerView.trackingScrollView = nil;
+  self.appBar.appBarViewController.headerView.trackingScrollView = nil;
```

Comment:
Shouldn't we replace the MDCAppBar property in this class?

Reply:
We still need to keep a reference to the app bar instance until we drop the MDCAppBar API.

I could add an extra property to store the app bar view controller, but it's a layer of indirection that isn't public and I worry doing so will make the internal implementation more complex.